### PR TITLE
Onfastpath

### PR DESCRIPTION
**Pull request by eeckstein**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
What is this for?

**Comment by eeckstein** [comment]
Basically it's to tell the inliner that such a code is worth inlining. It's similar to @inline(__always) but not as strict and it's effect is not limited to the containing function but also to it's callers (see also my comment on the first commit).

**Comment by jckarter** [comment]
Makes sense, thanks. Is it worth giving the builtin an adjustable "bias" parameter, to adjust how "hot" the path is?

**Comment by eeckstein** [comment]
It's definitely possible to add a bias parameter, and it's easy to implement. But this does not answer the question if it's worth. I suggest we add it as soon as there is a real need for it.

**Comment by jckarter** [comment]
Fair enough.

File: stdlib/public/core/Builtin.swift [hunk 1]
```diff
@@ -286,6 +286,13 @@ public func _slowPath<C : Boolean>(x: C) -> Bool {
   return _branchHint(x.boolValue, expected: false)
 }
 
+/// Optimizer hint that the code where this function is called is on the fast
+/// path.
+@_transparent
+public func _onFastPath() -> () {
```
Comment (gribozavr) [inline stdlib/public/core/Builtin.swift]: The `-> ()` part seems to be redundant...

Comment (eeckstein) [inline stdlib/public/core/Builtin.swift]: I wrote too much sil in the last time...
Thanks, I'll remove it

**Pull request merged**